components/server/src/orgs/organization-service.ts

```diff
@@ -658,6 +639,45 @@ export class OrganizationService {
         return this.toSettings(await this.teamDB.setOrgSettings(orgId, settings, mergeSettings));
     }
 
+    async getSettingsWithResolvedWelcomeMessage(userId: string, orgId: string): Promise<OrganizationSettings> {
+        const settings = await this.getSettings(userId, orgId);
+        return this.resolveWelcomeMessage(settings, orgId);
```

Comment:
IMO the resolution of the welcomeMessage should never fail or block the retrieval of the org settings. Rather, we should return `undefined` for avatarUrl.

In general it might make sense to only extract the `resolveFeaturedMemberAvatarURL` functionality, and keep the rest where it was.

Reply:
Also, this method lags a comment. :handshake:

Reply:
Addressed in [`d9e1cf2` (#20626)](https://github.com/gitpod-io/gitpod/pull/20626/commits/d9e1cf2d375af659a24324f6e360bcd44f1affd9)